SU2_CFD/src/fluid/CUserDefinedTCLib.cpp

```diff
@@ -0,0 +1,1667 @@
+/*!
+ * \file CUserDefinedTCLib.cpp
+ * \brief Source of user defined 2T nonequilibrium gas model.
+ * \author W. Maier, C. Garbacz
+ * \version 7.0.5 "Blackbird"
+ *
+ * SU2 Project Website: https://su2code.github.io
+ *
+ * The SU2 Project is maintained by the SU2 Foundation
+ * (http://su2foundation.org)
+ *
+ * Copyright 2012-2020, SU2 Contributors (cf. AUTHORS.md)
+ *
+ * SU2 is free software; you can redistribute it and/or
+ * modify it under the terMolarMass of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * SU2 is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with SU2. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "../../include/fluid/CUserDefinedTCLib.hpp"
+
+CUserDefinedTCLib::CUserDefinedTCLib(const CConfig* config, unsigned short val_nDim, bool val_viscous): CNEMOGas(config){
+
+  bool init_err;
+  unsigned short maxEl = 0;
+  su2double mf;
+
+  /*--- Allocate vectors for gas properties ---*/
+  nElStates.resize(nSpecies,0);
+  Wall_Catalycity.resize(nSpecies,0.0);
+  CharVibTemp.resize(nSpecies,0.0);
+  RotationModes.resize(nSpecies,0.0);
+  Enthalpy_Formation.resize(nSpecies,0.0);
+  Ref_Temperature.resize(nSpecies,0.0);
+  Diss.resize(nSpecies,0.0);
+  Blottner.resize(nSpecies,3) = su2double(0.0);
+  Omega00.resize(nSpecies,nSpecies,4,0.0);
+  Omega11.resize(nSpecies,nSpecies,4,0.0);
+
+  if(viscous){ 
+    MolarFracWBE.resize(nSpecies,0.0);
+    phis.resize(nSpecies,0.0);
+    mus.resize(nSpecies,0.0);
+  }
+
+
+  Kind_GasModel = config->GetKind_GasModel();
+
+  switch (Kind_GasModel) {
+
+  case N2:
+    /*--- Check for errors in the initialization ---*/
+    init_err = false;
+    if (nSpecies != 2) {
+      cout << "CONFIG ERROR: nSpecies mismatch between gas model & gas composition" << endl;
```

Comment:
These very long cases could be moved to some member function "one per case", it is usually good to have small functions, especially constructors.